### ISSUE TITLE
3.10 upgrade fails converting master config

### ISSUE THREAD

**Issue by clcollins** [openshift/openshift-ansible#9446]
#### Description

During upgrade from 3.9 to 3.10, upgrade fails attempting to convert master config to new format, at step:

`TASK [openshift_control_plane : Migrate old master env file to new location]`

Critically, attempting to re-try the upgrade after that point fails, because the master is no longer running:

`fatal: [openshift-master-dev-01]: FAILED! => {"changed": false, "msg": "Configmap for node-config-master must be present"}`

We are unsure how to recover from this, cannot get any further than this step after the initial failure, and the cluster is currently totally down.

##### Version

```
$ ansible --version
ansible 2.6.1
$ git describe
openshift-ansible-3.10.27-2-32-g8b7bbde6e
```

##### Steps To Reproduce
1. Attempt upgrade: `playbooks/byo/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml
`

##### Expected Results
Upgrade of control plane to succeed

```
Example command and output or error messages
```

##### Observed Results

Control plane upgrade fails at this step:

```
TASK [openshift_control_plane : Migrate old master env file to new location] ****************************
fatal: [openshift-master-dev-01]: FAILED! => {"changed": false, "module_stderr": "Shared connection to openshift-master-dev-01 closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_ofJYPd/ansible_module_master_env_config_migrate.py\", line 234, in <module>\r\n    main()\r\n  File \"/tmp/ansible_ofJYPd/ansible_module_master_env_config_migrate.py\", line 230, in main\r\n    run_module()\r\n  File \"/tmp/ansible_ofJYPd/ansible_module_master_env_config_migrate.py\", line 223, in run_module\r\n    create_file(src, dest)\r\n  File \"/tmp/ansible_ofJYPd/ansible_module_master_env_config_migrate.py\", line 197, in create_file\r\n    config.readfp(open(src))\r\n  File \"/usr/lib64/python2.7/ConfigParser.py\", line 324, in readfp\r\n    self._read(fp, filename)\r\n  File \"/tmp/ansible_ofJYPd/ansible_module_master_env_config_migrate.py\", line 115, in _read\r\n    inline_prefixes = {p: -1 for p in self._inline_comment_prefixes}\r\nAttributeError: SectionlessParserOld instance has no attribute '_inline_comment_prefixes'\r\n", "msg": "MODULE FAILURE", "rc": 1}
```

##### Additional Information

_operating system_

 rhel atomic host  7.5.1.1

_inventory file_

```
{
  "_meta": {
    "hostvars": {
      "openshift-master-dev-01": {
        "openshift_schedulable": true,
        "openshift_node_group_name": "node-config-master"
      },
      "openshift-node-dev-01": {
        "openshift_schedulable": true,
        "openshift_node_group_name": "node-config-infra"
      },
      "openshift-node-dev-02": {
        "openshift_schedulable": true,
        "openshift_node_group_name": "node-config-infra"
      },
      "openshift-node-dev-03": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-04": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-05": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-06": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-07": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-08": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-09": {
        "openshift_node_group_name": "node-config-compute"
      },
      "openshift-node-dev-10": {
        "openshift_node_group_name": "node-config-compute"
      }
    }
  },
  "OSEv3": {
    "children": [
      "masters",
      "nodes",
      "etcd"
    ],
    "vars": {
      "openshift_node_groups": [
        {
          "name": "node-config-master",
          "labels": [
            "node-role.kubernetes.io/master=true"
          ],
          "edits": []
        },
        {
          "name": "node-config-infra",
          "labels": [
            "node-role.kubernetes.io/infra=true"
          ],
          "edits": []
        },
        {
          "name": "node-config-compute",
          "labels": [
            "node-role.kubernetes.io/compute=true"
          ],
          "edits": []
        }
      ],
      "containerized": true,
      "deployment_type": "origin",
      "dynamic_volumes_check": false,
      "openshift_cloudprovider_kind": "vsphere",
      "openshift_cloudprovider_vsphere_datacenter": "Main Datacenter",
      "openshift_cloudprovider_vsphere_datastore": "UNI-ATC-01-BRONZE-OPENSHIFT",
      "openshift_cloudprovider_vsphere_folder": "REDACTED",
      "openshift_cloudprovider_vsphere_host": "vmware-test",
      "openshift_cloudprovider_vsphere_password": "REDACTED",
      "openshift_cloudprovider_vsphere_username": "os-disk-prov",
      "openshift_enable_service_catalog": true,
      "openshift_hosted_registry_routehost": "registry",
      "openshift_logging_install_logging": true,
      "openshift_logging_storage_kind": "dynamic",
      "openshift_master_cluster_public_hostname": "REDACTED",
      "openshift_master_cluster_hostname": "openshift-master-dev-01",
      "openshift_master_default_subdomain": "REDACTED",
      "openshift_master_identity_providers": [
        {
          "name": "openshift-dev",
          "challenge": "false",
          "login": "true",
          "mappingMethod": "claim",
          "kind": "OpenIDIdentityProvider",
          "clientID": "openshift_dev",
          "clientSecret": "REDACTED",
          "claims": {
            "id": [
              "sub"
            ],
            "preferredUsername": [
              "sub",
              "email"
            ],
            "name": [
              "nickname",
              "given_name",
              "name",
              "sub"
            ],
            "email": [
              "email",
              "sub"
            ]
          },
          "urls": {
            "authorize": "https://oauth/oidc/authorize",
            "token": "https://oauth/oidc/token",
            "userInfo": "https://oauth/oidc/userinfo"
          }
        }
      ],
      "openshift_master_named_certificates": [
        {
          "cafile": "REDACTED-ca.crt",
          "certfile": "REDACTED.combined.crt",
          "keyfile": "REDACTED.key"
        }
      ],
      "openshift_master_overwrite_named_certificates": true,
      "openshift_provisioners_install_provisioners": true,
      "openshift_release": "v3.10",
      "openshift_rolling_restart_mode": "system",
      "openshift_version": "3.10.0",
      "openshift_web_console_image_name": "web-console",
      "os_firewall_use_firewalld": false,
      "os_sdn_network_plugin_name": "redhat/openshift-ovs-multitenant",
      "template_service_broker_install": true,
      "openshift_prometheus_namespace": "openshift-metrics",
      "openshift_prometheus_node_selector": {
        "region": "infra"
      }
    }
  },
  "all": {
    "vars": {
      "openshift_disable_check": "disk_availability"
    }
  },
  "etcd": {
    "hosts": [
      "openshift-master-dev-01"
    ]
  },
  "masters": {
    "hosts": [
      "openshift-master-dev-01"
    ]
  },
  "nodes": {
    "hosts": [
      "openshift-master-dev-01",
      "openshift-node-dev-01",
      "openshift-node-dev-02",
      "openshift-node-dev-03",
      "openshift-node-dev-04",
      "openshift-node-dev-05",
      "openshift-node-dev-06",
      "openshift-node-dev-07",
      "openshift-node-dev-08",
      "openshift-node-dev-09",
      "openshift-node-dev-10"
    ]
  }
}
```

**Comment by clcollins** [comment]
@michaelgugino: Apologies if I'm out of line by tagging you - I just saw your name in the file where we're having trouble.  Please feel free to ignore if you're not/no longer responsible, and thanks for any assistance.

The issue appears to be with configparser.RawConfigParser missing `_inline_comment_prefixes`.

**Comment by michaelgugino** [comment]
@sdodson need to make sure we don't break 3.10.z

**Comment by michaelgugino** [comment]
@clcollins remove all comments from /etc/sysconfig/*-master-api

**Comment by sdodson** [comment]
@clcollins To manually migrate them the main thing to be aware of is that /etc/origin/master/master.env must have all variables that contain whitespace quoted. In /etc/sysconfig/atomic-openshift-master-api they're likely not quoted. Otherwise there shouldn't be any complex transformation required.

Even if /etc/sysconfig/atomic-openshift-master-api has been removed there should still be some backups of previous versions of the file to use as a starting point.

**Comment by clcollins** [comment]
@sdodson, @michaelgugino: thank you for your help.  I manually moved the config file to master.env, removed comments and added quotes to the strings with space, and (after monkeying around with the master-config.yaml to back out a change from the upgrade that prevented me from running the playbook again) am back on track and getting further.

**Comment by michaelgugino** [comment]
Fix merged in 3.10 branch.
